ceph/mds.go

```diff
@@ -0,0 +1,173 @@
+package ceph
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strconv"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	cephCmd                      = "/usr/bin/ceph"
+	mdsBackgroundCollectInterval = 5 * time.Minute
+)
+
+const (
+	MDSModeDisabled   = 0
+	MDSModeForeground = 1
+	MDSModeBackground = 2
+)
+
+type mdsStat struct {
+	FSMap struct {
+		Filesystems []struct {
+			MDSMap struct {
+				FSName string `json:"fs_name"`
+				Info   map[string]struct {
+					GID   uint   `json:"gid"`
+					Name  string `json:"name"`
+					Rank  int    `json:"rank"`
+					State string `json:"state"`
+				} `json:"info"`
+			} `json:"mdsmap"`
+		} `json:"filesystems"`
+	} `json:"fsmap"`
+}
+
+// runMDSStat will run mds stat and get all info from the MDSs within the ceph cluster.
+func runMDSStat(config, user string) ([]byte, error) {
+	var (
+		out []byte
+		err error
+	)
```

Comment:
Named returns?

Reply:
Just following the convention here across this repo. :)  Named returns can also get a little confusing at times so only when it's needed, maybe it's justified here because it's a small enough function, but we can make a change across the repo if we think that's a compelling reason to do so.